tardis/visualization/tools/convergence_plot.py

```diff
@@ -0,0 +1,360 @@
+        self.luminosity_plot = None
+
+        if "colorscale" in kwargs:
+            self.colorscale = transition_colors(
+                kwargs["colorscale"], iterations=self.iterations
+            )
+        else:
+            self.colorscale = transition_colors(iterations=self.iterations)
+
+        if "plasma_plot_config" in kwargs:
+            if kwargs["plasma_plot_config"] != {}:
+                self.plasma_plot_config = kwargs["plasma_plot_config"]
+
+        if "luminosity_plot_config" in kwargs:
+            if kwargs["luminosity_plot_config"] != {}:
+                self.luminosity_plot_config = kwargs["luminosity_plot_config"]
+
+    def fetch_data(self, name=None, value=None, item_type=None):
+        """
+        This allows user to fetch data from the Simulation class.
+        This data is stored and used when an iteration is completed.
+
+        Parameters
+        ----------
+        name : string
+            name of the data
+        value : string or array
+            string or array of quantities
+        item_type : string
+            either iterable or value
+
+        """
+        if item_type == "iterable":
+            self.iterable_data[name] = value
+        if item_type == "value":
+            self.value_data[name].append(value)
+
+    def create_plasma_plot(self):
+        """
+        Creates an empty plasma plot.
+        The default layout can be overridden by passing plasma_plot_config dictionary in the run_tardis function.
+        """
+        fig = go.FigureWidget().set_subplots(rows=1, cols=2, shared_xaxes=True)
+        fig.add_scatter(row=1, col=1)
+        fig.add_scatter(row=1, col=2)
+        fig = fig.update_layout(
+            xaxis={
+                "tickformat": "g",
+                "title": r"$Shell~~Velocity$",
+            },
+            xaxis2={
+                "tickformat": "g",
+                "title": r"$Shell~~Velocity$",
+                "matches": "x",
+            },
+            yaxis={
+                "tickformat": "g",
+                "title": r"$T_{rad}\ [K]$",
+                "tickmode": "auto",
+                "nticks": 15,
+            },
+            yaxis2={
+                "tickformat": "g",
+                "title": r"$W$",
+                "tickmode": "auto",
+                "nticks": 15,
+            },
+            height=450,
+            legend_title_text="Iterations",
+            margin=dict(l=10, r=135, b=25, t=25, pad=0),
+        )
+
+        # allows overriding default layout
+        if hasattr(self, "plasma_plot_config"):
+            self.override_plot_parameters(fig, self.plasma_plot_config)
+        self.plasma_plot = fig
+
+    def create_luminosity_plot(self):
+        """
+        Creates an empty luminosity plot.
+        The default layout can be overridden by passing luminosity_plot_config dictionary in the run_tardis function.
+        """
+        line_colors = ["orangered", "lightseagreen", "indigo"]
```

Comment:
Is it documented how can one override these colors?

Reply:
are these colours colour-blind friendly? the defaults should be suitable for all users. I think red/green is a no-go, and here we have orangered/green?

Reply:
^

Reply:
I'm afraid the colors are not color blind friendly. I will change that soon. These colors can be changed by passing a list of them or the name of the color in the `luminosity_colorscale` option. 
Something like this:
```py
sim = run_tardis(
    "tardis_example.yml",
    luminosity_colorscale = "viridis", 
)
```
Or this:
```py
sim = run_tardis(
    "tardis_example.yml",
    luminosity_colorscale = [
        'rgb(229, 134, 6)',
        'rgb(93, 105, 177)',
        'rgb(82, 188, 163)',
        'rgb(153, 201, 69)',
        'rgb(204, 97, 176)',
    ],
)
```

Reply:
Functionality added in https://github.com/tardis-sn/tardis/commit/38bcb465adf12322816f3194badb0fd3ca525a11, added documentation in https://github.com/tardis-sn/tardis/commit/d26efd54ea35a0cc121ab96612e0df1d8bd8f94e